decompiler/util/widget.py

```diff
@@ -174,11 +174,9 @@ def wrapped(self, *args, **kwargs):
 
             return wrapped
 
-    def __init__(self, parent: ViewFrame, name: str):
+    def __init__(self, data: BinaryView):
         """Initialize dock widget"""
-        self._is_widget_visible = False
-        QWidget.__init__(self, parent)
-        DockContextHandler.__init__(self, self, name)
+        QWidget.__init__(self)
         self._current_function: Optional[Function] = None
         self._current_view: Optional[BinaryView] = None
         self._current_frame: Optional[ViewFrame] = None
```

Comment:
I've been thinking that it could potentially more clean to model these fields as a separate class. Any thoughts on this?
e.g.

```python
class BinaryninjaState:
    def __init__(self, function, view, frame):
        ...
    @classmethod
    def get_current(cls):
        buffer = self.get_empty()
        buffer.update()
        return buffer

    def get_empty(cls):
        return cls(None, None. None)
    def update(self):
        pass
```

Reply:
Potentially, this could also isolate the notifications